src/avram/criteria.cr

```diff
@@ -16,29 +16,70 @@ class Avram::Criteria(T, V)
     rows
   end
 
-  # :nodoc:
-  def eq(_value : Nil)
-    {{ raise "Use `nilable_eq` instead of `eq` if trying to check against a nilable value." }}
-  end
-
   def eq(value)
+    check_just_nil!(typeof(value))
+    check_nilable!(value)
     perform_eq(value)
   end
 
+  private def check_nilable!(_value : Nil)
+    {% raise <<-ERROR
+
+      The 'eq' method cannot compare values that may be 'nil'
+
+      Try this...
+
+        ▸ Use 'nilable_eq'. This method allows checking against nilable values
+        ▸ Ensure the value is not nilable by wrapping it in an 'if' or using 'not_nil!'
+
+      ERROR
+    %}
+  end
+
+  private def check_nilable!(_value)
+    # carry on
+  end
+
+  private def check_just_nil!(_type_of_value : Nil.class)
+    {% raise <<-ERROR
+
+      To check if a column is 'nil' use these methods instead...
+
+        ▸ 'is_nil'
+        ▸ 'is_not_nil'
+
+      ERROR
+    %}
+  end
+
+  private def check_just_nil!(_type_of_value)
+    # carry on
+  end
+
   private def perform_eq(value)
     add_clause(Avram::Where::Equal.new(column, V::Lucky.to_db!(value)))
     rows
   end
 
   def nilable_eq(value)
+    check_just_nil!(typeof(value))
+
     if value.nil?
-      add_clause(Avram::Where::Null.new(column))
-      rows
```

Comment:
what was `rows` doing here, and is it still needed?

Reply:
I'm not sure if it is still needed. I'm getting a spec failure that may be related. Investiigating that now

Reply:
Turns out this was being subtly done by the `add_clause` method which returned `rows`. Instead I've removed that and explicitly return `rows` now. That's what allows chaining criteria

Reply:
Cancel that. This breaks all the other criteria. This will be a pain to always add so I'll change it back lol